### DISCUSSION TITLE
v20.1.0-rc7

### DISCUSSION THREAD

**Discussion by fballiano** [OpenMage/magento-lts#3409] [Releases]
This is gonna be the last RC release before tagging v20.1.0/v19.5.0 as stable. This will be done in about a week if no major problem is discovered. In this timeframe nothing new will be merged in OpenMage's repository, unless it's an important bugfix for this specific release. See you soon with the biggest release of OpenMage ever! 🎉

## You should absolutely know
Since the approval of [our second RFC - release schedule](https://github.com/OpenMage/rfcs/blob/main/accepted/0002-release-schedule.md)  OpenMage 19.x enters and "patch only" state, it will be maintained for two more years as promised but only _significant security patches or regression fixes_ will be ported to v19, every other development (and we have many) will be focused on v20+.

We encourage everybody to upgrade to v20, it is our latest and greatest and deserves the bit of work necessary for the upgrade (ask your developer/agency, don't do it yourself).

## Release highlights

This is a big release, that's why we decided to move away from the 20.0.x versioning and go to 20.1.x. Since a lot of changes could have some impact on current installations we decided to release some "rc" versions before the official 20.1.0. Tests are more than welcome now but be extra careful with production environment.

What's most important is the removal of all the 3rd party libraries (phpseclib, mcrypt_compat, Cm_RedisSession, Cm_Cache_Backend_Redis and Pelago_Emogrifier and Zend Framework) form our repository, they are now imported via composer. This was an important step to clean up and modernise our code.  
If you want to clean up your project from all the files that were removed during this release cycle, please [check this link](https://github.com/OpenMage/magento-lts/discussions/3170#discussioncomment-5722151) for the complete list.

Don't worry though, if you've always installed OpenMage extracting the zip file, starting from this release you'll find a new zip file attached to the release itself, we build this zip adding all of the old 3rd party libraries so that you will not have to migrate to composer or use composer at all.

Also:
- PHP 7.4 is now the minimum required version and 8.2 is now supported.
- The M1 legacy themes have been moved to an external repository since it's old (and mostly unused) code.
- A great improvement to EAV config cache has been added to v20.
- Support for Google Analytics 4 and Google Tag Manager was added with enhanced ecommerce data (which historically was always provided by 3rd party extensions).
- Possibility to set backend locale per every admin user was added.
- "pub/" folder structure was introduced (optional), to have extra security for your installation
- Massive performance improvement in complex SQL queries (#2210)
- Native WebP support
- Many, many smaller bugfix and improvements

Things you should be aware of:
- Support for Google Analytics3 and Universal Analytics were removed.
- Support for UPS CGI API (which was shut down in May 2023) was removed.
- Old inactive quotes are now actually purged from the database (#1489).

Known issues:
- The "pub/" folder structure won't work in composer managed projects, because the composer-magento plugin can't manage symlinks inside "pub/" correctly (#1210).

## Changelog
* Added DDEV command to install OM (incl. sample data) by @sreichel in https://github.com/OpenMage/magento-lts/pull/3248
* Allowed customization of the checkout page identifiers for the Mage_GoogleAnalytics module by @fballiano in https://github.com/OpenMage/magento-lts/pull/3363
* Bump phpstan/phpstan from 1.10.21 to 1.10.22 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3359
* Removed unexisting branch alias from composer.json by @fballiano in https://github.com/OpenMage/magento-lts/pull/3364
* Clear config cache after maintenance check for DB Update by @Hanmac in https://github.com/OpenMage/magento-lts/pull/3365
* Added hidden element before multiselect form elements in adminhtml by @Hanmac in https://github.com/OpenMage/magento-lts/pull/3352
* Added decimal/thousand separator specification when sending Analytics4 data by @fballiano in https://github.com/OpenMage/magento-lts/pull/3366
* Added DDEV command to create/update the administrator account by @ADDISON74 in https://github.com/OpenMage/magento-lts/pull/3256
* New feature to repopulate product form in frontend on errors. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3367
* Bump phpstan/phpstan from 1.10.22 to 1.10.25 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3372
* Bump phpseclib/phpseclib from 3.0.20 to 3.0.21 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3369
* Bump friendsofphp/php-cs-fixer from 3.20.0 to 3.21.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3370
* Bump magento-ecg/coding-standard from 4.5.2 to 4.5.4 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3371
* Converted line ending of some files in order to avoid warnings from git by @fballiano in https://github.com/OpenMage/magento-lts/pull/3368
* Resized panels height in "attribute set" backend view by @fballiano in https://github.com/OpenMage/magento-lts/pull/3247
* Fixed $address is null in AddressController. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3373
* Fixed exception in Dob.php during customer registration. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3374
* Better clearing of session messages after display by @kyrena in https://github.com/OpenMage/magento-lts/pull/3326
* Do not reindex catalog_product_flat for disabled stores by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3251
* Mage_Report: Migrated hardcoded cron expressions to configurable fields by @Sekiphp in https://github.com/OpenMage/magento-lts/pull/1869
* Mage_Eav_Model_Attribute_Data_Date::extractValue() should not return false by @fballiano in https://github.com/OpenMage/magento-lts/pull/3268
* Fixed customer not receiving order notification email when the email … by @kiatng in https://github.com/OpenMage/magento-lts/pull/3290
* Removed double save on quote in Mage_Checkout_CartController::ajaxUpdateAction() by @fballiano in https://github.com/OpenMage/magento-lts/pull/3378
* Rewrote part of GA4 to support add/remove quantity from cart and cart_updates by @fballiano in https://github.com/OpenMage/magento-lts/pull/3377
* Fixed implicit conversion from float to int in Mage_SalesRule_Model_Validator by @kyrena in https://github.com/OpenMage/magento-lts/pull/3345
* Rewrote afterSave of product and category image attributes by @luigifab in https://github.com/OpenMage/magento-lts/pull/3301
* Bump friendsofphp/php-cs-fixer from 3.21.0 to 3.22.0 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3385
* Bump phpunit/phpunit from 9.6.9 to 9.6.10 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3386
* Fixed passing null for number_format is deprecated for downloadable products by @fballiano in https://github.com/OpenMage/magento-lts/pull/3382
* Allowed to upload images in WebP format by @empiricompany in https://github.com/OpenMage/magento-lts/pull/3384
* Removed phpstan.dist.issues.neon by @sreichel in https://github.com/OpenMage/magento-lts/pull/3257
* Fixed issue #3392 clearInstance() method in Mage_Catalog_Model_Product. by @kiatng in https://github.com/OpenMage/magento-lts/pull/3395
* Bump phpstan/phpstan from 1.10.25 to 1.10.26 by @dependabot in https://github.com/OpenMage/magento-lts/pull/3402
* Fixed a bug with UPS that prevented saving the shipping methods page from backend by @fballiano in https://github.com/OpenMage/magento-lts/pull/3397
* Fixed documentation block typo on Mage::addObserver by @denisahac in https://github.com/OpenMage/magento-lts/pull/3404
* Fixed truecolor detection for WebP images by @fballiano in https://github.com/OpenMage/magento-lts/pull/3403
* Fixed a bug with overlapping menu items in backend by @seifer7 in https://github.com/OpenMage/magento-lts/pull/3396

## New Contributors
* @Hanmac made their first contribution in https://github.com/OpenMage/magento-lts/pull/3365
* @denisahac made their first contribution in https://github.com/OpenMage/magento-lts/pull/3404
* @seifer7 made their first contribution in https://github.com/OpenMage/magento-lts/pull/3396

**Full Changelog**: https://github.com/OpenMage/magento-lts/compare/v20.1.0-rc6...v20.1.0-rc7

<hr /><em>This discussion was created from the release <a href='https://github.com/OpenMage/magento-lts/releases/tag/v20.1.0-rc7'>v20.1.0-rc7</a>.</em>

**Comment by elidrissidev** [comment]
One other issue that's worth adding to "Known issues" which I mentioned previously and I think I also saw it on Discord recently, is the issue with overriding `Zend_*` classes by putting them inside `app` folder.

**Reply by fballiano** [reply]
can you link it to me?

**Reply by elidrissidev** [reply]
https://github.com/OpenMage/magento-lts/pull/2861#issuecomment-1367414262

I'll post the Discord one later, can't use it now.

**Reply by fballiano** [reply]
actually in the link that's linked (sorry) to know all of the folders that can be deleted there's also "app/code/core/Zend" and I kinda wouldn't consider this an "issue", it's just part of this change. I tend to think of an issue as something we will fix later but here there's actually nothing to fix no?

**Reply by elidrissidev** [reply]
Here's a recent one from Discord: https://discord.com/channels/408199701196963842/748568096989511720/1141673669916098560

**Comment by alexherbs** [comment]
@fballiano Will there be also a further RC release for v19? Or did I miss something?

**Reply by fballiano** [reply]
There were no changes since the last rc 😉

**Reply by alexherbs** [reply]
@fballiano Okay thank you, i'm just a little confused because there are some much changes in rc6 and rc7. But its okay. Thank you

**Reply by fballiano** [reply]
There were, but only for the v20 branch since v19 is in freeze mode now, so only patch level changes will be backported
